pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,30 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return operationContinuing(powerRequeueDelay)
+			default:
+				// For other errors continue with the delete
+				if p.host.Status.ErrorCount > maxErrorRetryCount {
+					result.Dirty = false
+					p.log.Info("could not power off host before removal")
+					operationComplete()
```

Comment:
missing `return`

Reply:
returning here would mean we would not proceed to nodes.Delete(), right?

Reply:
Sorry I was probably too concise, my point is that the semantic functions like `operationComplete` are meant as a wrapper to package a `provisioner.Result` and an `error`, so its return value must be used somehow

Reply:
Removed the operationComplete() here because it is premature, imo. We want to proceed to the delete of the node.

Reply:
I think that the threshold check is not in the right position. In case of `hardPowerOff` error another reconcile will be invoked (thanks to the `operationFailed`), so the (increased) `host.Status.ErrorCount` could be evaluated only in the next loop. This means that the check must be performed _before_ `hardPowerOff`, ie something like:

`if ironicNode.PowerState == powerOn && p.host.Status.ErrorCount < maxPowerOffRetryCount {
...
}`

Reply:
Ah! Got it! Fixing it in my next version.